File: leetcode/go/11 Container With Most Water.go
func maxArea(height []int) int {
    res, i, j := 0, 0, len(height) - 1
    for;i < j;{
        res = max(res, min(height[i], height[j])*(j - i))
        
        if(height[i] < height[j]){
            i++
        }else{
            j--
        }
    }
    return res
}

func max(val1 int, val2 int) int {
    if(val1 > val2){
        return val1
    }else{
        return val2
    }
}

func min(val1 int, val2 int) int {
    if(val1 < val2){
        return val1
    }else{
        return val2
    }
}

File: leetcode/ts/11 Container With Most Water.ts
function maxArea(height: number[]): number {
    let left : number = 0;
    let right : number = height.length - 1;
    let capacity : number = 0;
    
    while(left < right){
        capacity = Math.max(capacity, Math.min(height[left], height[right]) * (right - left))
        height[left] < height[right] ? ++left : --right
    }
    return capacity
};